solutions/USACO open19-milk-factory.mdx

```diff
@@ -153,3 +159,49 @@ int main(){
 
 </CPPSection>
 </LanguageSection>
+
+## Solution 2
+
+This solution follows the logic outlined in the official solution.
+
+## Implementation
+
+**Time Complexity**: $\mathcal{O}(N)$
+
+<LanguageSection>
+<PySection>
+
+```py
+# Take in the input using standard Python file i/o system
+data = open('factory.in').read().strip().split('\n')
+n = int(data[0])
+conveyer_belts = [list(map(int, data[i + 1].split(' '))) for i in range(n - 1)]
+
+# Create a list holding all factories that have an outgoing conveyer belts
+outgoing_locations = [0]*max(outgoing)
+
+for i in range(n-1):
+	outgoing_locations[conveyer_belts[i][0] - 1] += 1
+
+ans = -1
+counter = 0
+
+# Find out which station does not have any outgoing conveyer belts
+for i in range(n - 1):
+	if outgoing_locations[conveyer_belts[i][1] - 1] == 0:
+		if conveyer_belts[i][1] != ans:
```

Comment:
alternatively, you could just check all the stations in order so you wouldn't have to check if the station has already been seen, but this is fine